CodePushTelemetryManager.m

```diff
@@ -0,0 +1,125 @@
+            NSString *previousDeploymentKey = [self getDeploymentKeyFromStatusReportIdentifier:previousStatusReportIdentifier];
+            NSString *previousLabel = [self getVersionLabelFromStatusReportIdentifier:previousStatusReportIdentifier];
+            return @{
+                     @"appVersion": appVersion,
+                     @"previousDeploymentKey": previousDeploymentKey,
+                     @"previousLabelOrAppVersion": previousLabel
+                     };
+        } else {
+            // Previous status report was with a binary app version.
+            return @{
+                     @"appVersion": appVersion,
+                     @"previousLabelOrAppVersion": previousStatusReportIdentifier
+                     };
+        }
+    }
+    
+    return nil;
+}
+
++ (NSDictionary *)getRollbackReport:(NSDictionary *)lastFailedPackage
+{
+    return @{
+              @"package": lastFailedPackage,
+              @"status": DeploymentFailed
+            };
+}
+
++ (NSDictionary *)getUpdateReport:(NSDictionary *)currentPackage
+{
+    NSString *currentPackageIdentifier = [self getPackageStatusReportIdentifier:currentPackage];
+    NSString *previousStatusReportIdentifier = [self getPreviousStatusReportIdentifier];
+    if (currentPackageIdentifier) {
+        if (previousStatusReportIdentifier == nil) {
+            [self recordDeploymentStatusReported:currentPackageIdentifier];
+            return @{
+                     @"package": currentPackage,
+                     @"status": DeploymentSucceeded
+                     };
+        } else if (![previousStatusReportIdentifier isEqualToString:currentPackageIdentifier]) {
+            [self recordDeploymentStatusReported:currentPackageIdentifier];
+            if ([self isStatusReportIdentifierCodePushLabel:previousStatusReportIdentifier]) {
+                NSString *previousDeploymentKey = [self getDeploymentKeyFromStatusReportIdentifier:previousStatusReportIdentifier];
+                NSString *previousLabel = [self getVersionLabelFromStatusReportIdentifier:previousStatusReportIdentifier];
+                return @{
+                         @"package": currentPackage,
+                         @"status": DeploymentSucceeded,
+                         @"previousDeploymentKey": previousDeploymentKey,
+                         @"previousLabelOrAppVersion": previousLabel
+                         };
+            } else {
+                // Previous status report was with a binary app version.
+                return @{
+                         @"package": currentPackage,
+                         @"status": DeploymentSucceeded,
+                         @"previousLabelOrAppVersion": previousStatusReportIdentifier
+                         };
+            }
+        }
+    }
+    
+    return nil;
+}
+
++ (NSString *)getDeploymentKeyFromStatusReportIdentifier:(NSString *)statusReportIdentifier
```

Comment:
Since you're not sorting all methods together, what do you think about adding a "#pragma mark - Private methods" line above this group so it's clear what the two method groups are